provenancecheck.go

```diff
@@ -0,0 +1,8 @@
+package voucher
+
+// ProvenanceCheck represents a Voucher check that sets
+// trusted projects and build creators
+type ProvenanceCheck interface {
+	Check
+	SetTrustedBuilderIdentities(map[string]bool, map[string]bool)
```

Comment:
As I mentioned in my previous comment I do think it's worth having two separate functions for setting the trusted builders and identities. Could you re-split these out?

As well, the function names are very long, could we shorten them to:

```suggestion
	SetTrustedBuildCreators([]string)
	SetTrustedProjects([]string)
```
?

Reply:
Done.